jira/client.py

```diff
@@ -849,8 +850,10 @@ def add_attachment(
         Returns:
             Attachment
         """
+        close_attachment = False
         if isinstance(attachment, str):
             attachment: BufferedReader = open(attachment, "rb")  # type: ignore
+            close_attachment = True
```

Comment:
Let's put the cast earlier over here:

```suggestion
            attachment: BufferedReader = open(attachment, "rb")
            attachment = cast(BufferedReader, attachment)
            close_attachment = True
```

Then you won't need to cast later on.

You can blame me for just ignoring that type warning, sorry about that !

Reply:
It's a good idea. Thanks for your help.